### PR TITLE
Feature/output basename

### LINKED ISSUE #360: Add option to change output filename
The `routes.ts` output filename is currently hard coded:
https://github.com/lukeautry/tsoa/blob/7c9dbd0798340c04721dfdca3f2fd31d8be3aca8/src/routeGeneration/routeGenerator.ts#L26

I'd love to have an option to change this filename. I was going to make a PR for this, but the fact that such a trivial thing hasn't been implemented yet makes me wonder if there's an actual reason for it.

### PR DESCRIPTION
**Pull request by lucasrcosta**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [x] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

This is essentially a trimmed revamp of https://github.com/lukeautry/tsoa/pull/589 by @edbrannin. I kept the relevant commits from his branch, rebased it in top of master, and fixed it.

closes #360

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@tsoa/cli",
   "description": "Build swagger-compliant REST APIs using TypeScript and Node",
-  "version": "3.3.0",
+  "version": "3.3.1",
```
Comment (WoH) [inline packages/cli/package.json:4]:
```suggestion
  "version": "3.3.0",
```

File: packages/cli/package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
   "author": "Luke Autry <lukeautry@gmail.com> (http://www.lukeautry.com)",
   "license": "MIT",
   "dependencies": {
-    "@tsoa/runtime": "^3.3.0",
+    "@tsoa/runtime": "^3.3.1",
```
Comment (WoH) [inline packages/cli/package.json:31]:
```suggestion
    "@tsoa/runtime": "^3.3.0",
```

File: packages/runtime/package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@tsoa/runtime",
   "description": "Build swagger-compliant REST APIs using TypeScript and Node",
-  "version": "3.3.0",
+  "version": "3.3.1",
```
Comment (WoH) [inline packages/runtime/package.json:4]:
```suggestion
  "version": "3.3.0",
```

File: packages/tsoa/package.json [hunk 4]
```diff
@@ -25,8 +25,8 @@
   "author": "Luke Autry <lukeautry@gmail.com> (http://www.lukeautry.com)",
   "license": "MIT",
   "dependencies": {
-    "@tsoa/cli": "^3.3.0",
-    "@tsoa/runtime": "^3.3.0"
+    "@tsoa/cli": "^3.3.1",
+    "@tsoa/runtime": "^3.3.1"
```
Comment (WoH) [inline packages/tsoa/package.json:29]:
```suggestion
    "@tsoa/cli": "^3.3.0",
    "@tsoa/runtime": "^3.3.0"
```

File: tests/package.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   "name": "tsoa-tests",
   "private": true,
   "description": "Build swagger-compliant REST APIs using TypeScript and Node",
-  "version": "3.3.0",
+  "version": "3.3.1",
```
Comment (WoH) [inline tests/package.json:5]:
```suggestion
  "version": "3.3.0",
```

File: packages/cli/src/module/generate-spec.ts [hunk 6]
```diff
@@ -7,6 +7,13 @@ import { SpecGenerator2 } from '../swagger/specGenerator2';
 import { SpecGenerator3 } from '../swagger/specGenerator3';
 import { fsExists, fsMkDir, fsWriteFile } from '../utils/fs';
 
+export const getSwaggerOutputPath = (swaggerConfig: ExtendedSpecConfig) => {
+  const ext = swaggerConfig.yaml ? 'yaml' : 'json';
+  const specFileBaseName = swaggerConfig.specFileBaseName || 'openapi';
```
Comment (WoH) [inline packages/cli/src/module/generate-spec.ts:12]:
```suggestion
  const specFileBaseName = swaggerConfig.specFileBaseName || 'swagger';
```

For Context: The previous PR was opened right before a major release (for type aliases), so I felt comfortable to make a backwards-incompatible change. If we have no other reason to make a major bump, I'd keep the current default name and merge in time for the next minor.

File: packages/runtime/src/config.ts [hunk 7]
```diff
@@ -58,6 +58,14 @@ export interface SpecConfig {
   host?: string;
 
   /**
+   * Base-name of swagger.json or swagger.yaml.
+   *
+   * @default: "openapi"
```
Comment (WoH) [inline packages/runtime/src/config.ts:63]:
```suggestion
   * @default: "swagger"
```

File: packages/cli/package.json (hunk 1 above)
Comment (WoH) [reply packages/cli/package.json:4]: Don't bump versions here.

**Comment by lucasrcosta** [comment]
I have removed the version bump commit, as reprimanded, and replaced the default name on the tests as well.

**Review by WoH — APPROVED** [review]

**Comment by WoH** [comment]
LGTM, thank you for the PR!

**Comment by lucasrcosta** [comment]
Kudos to @edbrannin 🎉

**Pull request merged**